fw/apm.h

```diff
@@ -62,21 +62,43 @@ static const unsigned int tfw_pstats_ith[] = {
 
 int tfw_apm_add_srv(TfwServer *srv);
 void tfw_apm_del_srv(TfwServer *srv);
+
+/*
+ * All blocks of the procedures listed below, separated by spaces, are not
+ * data-coupled. It can be said that these are independent submodules of the
+ * APM module.
+ */
+
+/* Procedures related to statistics (avg/min/max/percentiles).
+ * Configured by the 'apm_stats' directive.
+ */
 void tfw_apm_update(void *apmref, unsigned long jtstamp, unsigned long jrtime);
 int tfw_apm_stats(void *apmref, TfwPrcntlStats *pstats);
 int tfw_apm_stats_bh(void *apmref, TfwPrcntlStats *pstats);
 int tfw_apm_pstats_verify(TfwPrcntlStats *pstats);
-void tfw_apm_hm_srv_rcount_update(TfwStr *uri_path, void *apmref);
+
+/* Cache hits/misses statistics. */
 void tfw_apm_cache_inc_hits(void *apmref);
 void tfw_apm_cache_inc_misses(void *apmref);
 void tfw_apm_cache_stats(TfwCacheStat *stats, void *apmref);
+
+/*
+ * Although these procedures are used for health monitoring, they also collect
+ * some general server statistics. Therefore, they work even when health
+ * monitoring is turned off. In this case, server suspension and resumption
+ * will not work, and only overall response statistics for the entire uptime
```

Comment:
It's not particularly clear what you wanted to say. Perhaps, you can rephrase or elaborate?

Reply:
Done